cargo-shuttle/src/config.rs

```diff
@@ -320,3 +334,81 @@ impl RequestContext {
             .unwrap()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{path::PathBuf, str::FromStr};
+
+    use shuttle_common::project::ProjectName;
+
+    use crate::{args::ProjectArgs, config::RequestContext};
+
+    fn path_from_workspace_root(path: &str) -> PathBuf {
+        PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap())
+            .join("..")
+            .join(path)
+    }
+
+    #[test]
+    fn get_local_config_finds_name_in_shuttle_toml() {
+        let project_args = ProjectArgs {
+            working_directory: path_from_workspace_root("examples/axum/hello-world/"),
+            name: None,
+        };
+
+        let local_config = RequestContext::get_local_config(&project_args).unwrap();
+
+        // FIXME: make a helper for this?
+        let name = local_config
+            .as_ref()
+            .unwrap()
+            .name
+            .as_ref()
+            .unwrap()
+            .to_string();
+        assert_eq!(name, "hello-world-axum-app");
+    }
+
+    #[test]
+    fn fixme_running_in_src_subdir_finds_crate_but_fails_to_find_config() {
+        let project_args = ProjectArgs {
+            working_directory: path_from_workspace_root("examples/axum/hello-world/src"),
+            name: None,
+        };
+
+        let local_config = RequestContext::get_local_config(&project_args).unwrap();
+
+        // FIXME: make a helper for this?
+        let name = local_config
+            .as_ref()
+            .unwrap()
+            .name
+            .as_ref()
+            .unwrap()
+            .to_string();
+
+        // FIXME: this is not the intended behaviour. We should fix this.
+        // This should really be "hello-world-axum-app", as above.
+        assert_eq!(name, "hello-world");
```

Comment:
I guess I should do this as part of this PR? I think it's an existing bug, so I would usually fix it as a separate PR, to avoid things dragging out too long unmerged. If we want it fixed as part of this PR then I can look at it later in the week (requires re-jigging when we call out to `cargo metadata`, I think).

Reply:
We can keep this as a separate PR. Will just need to open an issue for it then.

Reply:
Done: https://github.com/shuttle-hq/shuttle/issues/163